src/filesystem/impls/lib/HTMLRewriter.js

```diff
@@ -0,0 +1,193 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, appshell, DOMParser */
+define(function (require, exports, module) {
+    "use strict";
+
+    var Content = require("filesystem/impls/lib/content");
+    var Log = require("filesystem/impls/lib/log");
+    var Async = require("filesystem/impls/lib/async");
+    var CSSRewriter = require("filesystem/impls/lib/CSSRewriter");
+
+    var Filer = appshell.Filer;
+    var Path = Filer.Path;
+
+    /**
+     * Rewrite all external resources (links, scripts, img sources, ...) to
+     * blob URL Objects from the fs.
+     */
+    function HTMLRewriter(fs, path, html) {
+        this.fs = fs;
```

Comment:
Same thing here, don't bother passing `fs` in as an arg, just get it from `Filer` on line 19.